olive/strategy/search_parameter.py

```diff
@@ -80,7 +80,7 @@ class Boolean(Categorical):
     >>> Boolean()
     """
 
-    def __init__(self):
+    def __init__(self, support: List[bool] = None):
         super().__init__([True, False])
```

Comment:
not pass default arrays, Instead, better pass None and let the Categorical handle Non

Reply:
Emm, Boolean() is used to directly create Categorical([True, False]). If we pass None, we can directly use Categorical.

(If the arguments `support` confuses you, I removed in latest commit as it is the zombie code.

Reply:
I mean 
![image](https://github.com/microsoft/Olive/assets/817030/2524af49-bc55-4c68-90ad-b77fd01c6a99)